### PR TITLE
Fix and reenable prance-based openapi spec validation

### PR DESCRIPTION
**Pull request by cognifloyd**
### Background

I'm working towards introducing [`pants`](https://www.pantsbuild.org/docs). One of the first things we will use it for is linting / reformatting (it will run black, flake8, etc). Validation of our openapi schema should happen more regularly, so we'll hook that up so pants runs it when it runs lint tools

While hooking that up, I noticed parts of the validation commented out. So, I searched through the git history to figure out when and why.
Basically, it's just tech debt that no one had time to address. So, this is a step in that direction.

### Overview

Fix and reenable prance-based openapi spec validation, but make our custom `x-api-model` validation optional as the spec is out-of-date.

This also fixes issues identified by the now-enabled spec validation.

Related: #3575 #3788

### CODE REVIEW

**Review by armab — APPROVED** [review]

File: st2common/st2common/cmd/validate_api_spec.py [hunk 1]
```diff
@@ -22,8 +22,8 @@
 from __future__ import absolute_import
 import os
 
+import prance
```
Comment (nzlosh) [inline st2common/st2common/cmd/validate_api_spec.py:25]: I understand the help `prance` offers with openapi, but I'm a little concerned with the activity of the project.  It claims to be tested against py36-py38 and hasn't received an update since 26 Nov 2021.  Perhaps this is a sign of its code maturity, but it would be unfortunate to fall into the same trap we did with `udatetime` where it became more of a hindrance than a help.  If you're of the opinion that the project is alive and active, I'm OK with adding it as a dependency.

Comment (nzlosh) [reply st2common/st2common/cmd/validate_api_spec.py:25]: OK, ignore me, it's not an addition, it's already a dependency, sorry for the noise.  But my concern still stands, we may need to look for an active project if `prance` has been abandoned.

**Review by nzlosh — APPROVED** [review]
LGTM

File: st2common/st2common/cmd/validate_api_spec.py (hunk 1 above)
Comment (cognifloyd) [reply st2common/st2common/cmd/validate_api_spec.py:25]: Agreed. There are a bunch of things I'd like to improve around the openapi spec. Getting rid of `prance` sounds great to me. We're also using openapi/swagger 2, which means we can't use a lot of newer tooling that only supports openapi 3+. So much to do...
